packages/gasket-redux/src/configure-make-store.js

```diff
@@ -4,37 +4,68 @@ import { createLogger } from 'redux-logger';
 import Log from '@gasket/log';
 import placeholderReducers from './placeholder-reducers';
 
+/**
+ * Compose the reducer
+ *
+ * @param {object} allReducers - Map of identifiers and reducers
+ * @param {function} [rootReducer] - Entry reducer to run before combined reducers
+ * @returns {function} reducer
+ * @private
+ */
+export function prepareReducer(allReducers, rootReducer) {
+  const combinedReducer = Object.keys(allReducers).length ? combineReducers(allReducers) : (f = {}) => f;
+
+  if (rootReducer) {
+    return (state, action) => {
+      const nextState = rootReducer(state, action);
+      // if state results are unchanged, continue to combined reducers
+      if (nextState === state) {
+        return combinedReducer(state, action);
+      }
+      return nextState;
+    };
+  }
+
+  return combinedReducer;
+}
 /**
  * Set up redux store configuration and return a makeStore function
  *
  * @param {Object} options - Options for create store
  * @param {Object.<String,Function>} options.reducers - Map of identifiers and reducers
+ * @param {function} [options.rootReducer] - Entry reducer to run before combined reducers
  * @param {Object} [options.initialState] - Initial redux state
  * @param {Array} [options.middleware] - Middleware
  * @param {Function[]} [options.enhancers] - Any additional enhancers
  * @param {Boolean} [options.logging] - logging is enabled by default. Passing false will disable logging completely.
  * @param {Function} [options.thunkMiddleware] - Optionally provide an extra argument for thunks
  * @param {Function} [postCreate] - Optional callback
- * @returns {function(*=): Store<{}>} makeStore
+ * @returns {MakeStore} makeStore
  */
 export default function configureMakeStore(
-  { reducers = {}, initialState = {}, middleware = [], enhancers = [f => f], logging = false, thunkMiddleware = thunk } = {},
+  {
```

Comment:
If `rootReducer` was given a default implementation, wouldn't that make the upgrade much easier? As is, you'd have to change your store code so this is a breaking change. Seems like a default shallow merge reducer (or deep if we switch to that) would be helpful here, since someone upgrading from an earlier version probably isn't even using store dispatches on the server side props handler.

Reply:
A default would only be needed/helpful if next-redux-wrapper@6 is installed and being used. I had started down that path but had concerns I mentioned in [this comment](https://github.com/godaddy/gasket/pull/173#discussion_r486776184). Having no `rootReducer` specified results in our current behavior. It's just a new option that can be used, and allows fulfillment of a requirement for next-redux-wrapper@6.

I'd like to keep `@gasket/redux` and `@gasket/plugin-redux` be `next` and `next-redux-wrapper` agnostic.